pkg/account/account.go

```diff
@@ -0,0 +1,71 @@
+package account
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	"go.uber.org/zap"
+
+	"github.com/archway-network/relayer_exporter/pkg/chain"
+	log "github.com/archway-network/relayer_exporter/pkg/logger"
+)
+
+type Account struct {
+	Address string `yaml:"address"`
+	Denom   string `yaml:"denom"`
+	ChainID string `yaml:"chainId"`
+	Balance math.Int
+}
+
+func GetBalances(accounts []Account, rpcs map[string]string) []Account {
+	num := len(accounts)
+
+	out := make(chan Account, num)
+	defer close(out)
+
+	for i := 0; i < num; i++ {
+		go func(i int) {
+			account, err := GetBalance(accounts[i], rpcs)
+			if err != nil {
+				out <- Account{}
+
+				log.Error(err.Error(), zap.Any("account", accounts[i]))
+
+				return
+			}
+			out <- account
+		}(i)
+	}
+
+	accountsWithBalance := []Account{}
+
+	for i := 0; i < num; i++ {
```

Comment:
Does this loop do anything?
Could this logic be done in the upper loop?

Reply:
Upper loop starts goroutines to fetch balances concurrently for many accounts and this loop is to wait for started earlier goroutines and gather results.